contracts/CDPosition.sol

```diff
@@ -22,6 +22,10 @@ contract CDPosition {
 
     mapping(uint256 => cdp) internal nftCDP;
 
+    uint256 internal _totalNftEntries;
+    uint256[] internal _nftIDArray; // list out all valid and live NFT ID
+    mapping(uint256 => uint256) internal _nftIDToArrayLocation;
```

Comment:
I am not sure exactly why you need this thing. What's "ArrayLocation"?

Why not having NFTID=>CDP struct mapping and just use `delete` to delete unneeded positions?

Reply:
!! This PR might  be reverted so no point in lingering over it !!

Deleting an NFT from NFTID => CDP is not what this attempt to solve. The array is needed for iterating over all open positions and handing out OUSD interest (need to iterate over all valid NFTID, which is what the array has).